### ISSUE TITLE
virtualenv shouldn't fatal if lock directory isn't writable

### ISSUE THREAD

**Issue by legoktm** [pypa/virtualenv#1640]
Traceback/reproduction steps:
https://gist.github.com/legoktm/533669e450d0f5f41838a4e9b858d774

By default the lock directory seems to be `~/.local/share/...`, but this is problematic in our setup, where our home directory is set to `/nonexistent`.

Wouldn't it be better to store temporary lock files in a temporary directory aka `/tmp` (and its equivalents on other OS's)?

Also, I don't expect the lack of a writable directory to fail so hard.  You can't even get `virtualenv --help` to work:
```
nobody@4b70ed23eebd:/src/apt-browser$ /src/apt-browser/.tox/.tox/bin/python -m virtualenv --help
FileNotFoundError: [Errno 2] No such file or directory: '/nonexistent/.local/share/virtualenv/py-info/20.0.4/2d7075eda06572014a3dfd447e6915f0dadc794767f53ba9bc4e3ed19eeb098d.lock'
```

**Comment by gaborbernat** [comment]
This would make sense 👍 I'll add it as an enhancement request.

**Comment by gaborbernat** [comment]
Hello, a fix for this issue has been released via virtualenv 20.0.6; see https://pypi.org/project/virtualenv/20.0.6/ (https://virtualenv.pypa.io/en/latest/changelog.html#v20-0-6-2020-02-26) . Please give a try and report back if your issue has not been addressed; if not, please comment here, and we'll reopen the ticket. We want to apologize for the inconvenience this has caused you and say thanks for having patience while we resolve the unexpected bugs with this new major release.
![thanks](https://media.tenor.co/images/1114067881329bb4218880eeb7ec34fd/tenor.gif)
